libs/go/aealite/protocols/dialogue.go

```diff
@@ -0,0 +1,300 @@
+	getRole() Role
+	getRules() RuleType
+	getMessageClass() AbstractMessage
+	isSelfInitiated() bool
+	getLastIncomingMessage() AbstractMessage
+	getLastOutgoingMessage() AbstractMessage
+	getLastMessage() AbstractMessage
+	isEmpty() bool
+	counterPartyForMessage(AbstractMessage) bool
+	isMessageBySelf(AbstractMessage) bool
+	isMessageByOther(AbstractMessage) bool
+	getMessage(MessageId) AbstractMessage
+	hasMessageId(MessageId) bool
+	update(AbstractMessage)
+	isBelongingToADialogue(AbstractMessage) bool
+	reply(Performative, Target) AbstractMessage
+	validateNextMessage(AbstractMessage) (bool, AbstractMessage)
+	basicValidations(AbstractMessage) (bool, AbstractMessage)
+	basicValidationInitialMessage(AbstractMessage) (bool, AbstractMessage)
+	basicValidationNonInitialMessage(AbstractMessage) (bool, AbstractMessage)
+	validateMessageTarget(AbstractMessage) string
+	validateMessageId(AbstractMessage) string
+	getMessageById(MessageId) AbstractMessage
+	getOutgoingNextMessageId() int
+	getIncomingNextMessageId() int
+	updateDIalogueLabel(DialogueLabel)
+	customValidation(AbstractMessage) (bool, string)
+	getStringRepresentation() string
+}
+
+type DialoguesInterface interface {
+	// initialize(dialogue DialogueLabel, endStates FrozenSet, _message_class InitialMessage, dialogueClass Dialogue, roleFromFirstMessage Role, keepTerminalStateDialogues bool)
+	isKeepDIaloguesInTerminalState() bool
+	selfAddress() Address
+	messageClass() AbstractMessage
+	dialogueClass() Dialogue
+	getDialoguesWithCounterParty(counterPArty Address) []Dialogue
+	isMessageBySelf(AbstractMessage) bool
+	isMessageByOther(AbstractMessage) bool
+	counterPartyFromMessage(AbstractMessage) Address
+	newSelfInitiatedDialogueReference() [2]string
+	create(counterParty Address, performative Performative, message SomeMessageType) (AbstractMessage, Dialogue)
+	createWithMessage(counterParty Address, intitialMessage AbstractMessage) Dialogue
+	createDialogue(counterParty Address, intitialMessage AbstractMessage) Dialogue
+	update(AbstractMessage) Dialogue
+	completeDialogueReference(AbstractMessage)
+	getDialogue(AbstractMessage) Dialogue
+	getLatestLabel(DialogueLabel) DialogueLabel
+	getDialogueFromLabel(DialogueLabel) Dialogue
+	createSelfInitiated(dialogueOpponentAddress Address, dialogueReference [2]string, role Role) Dialogue
+	createOpponentInitiated(dialogueOpponentAddress Address, dialogueReference [2]string, role Role) Dialogue
+	createInternal(incompleteDialogueLabel DialogueLabel, role Role, completeDialogueLabel DialogueLabel) Dialogue
+	generateDialogueNonce() string
+	setUpDialogueStorage()
+	tearDownDialogueStorage()
+}
+
+type AbstractMessageInterface interface {
+	initialize(diaglogReference [2]string, messagId int, target Target, performative Performative)
+	validPerformatives() []string
+	hasSender() bool
+	sender() Address
+	hasTo() bool
+	to() Address
+	dialogueReference() [2]string
+	messageId() MessageId
+	performative() Performative
+	target() int
```

Comment:
Let's declare a specific type for target

Reply:
Why is that needed? 
Target is equivalent to messageID in type (it is essentially messageID of another message). So make sense for both to have the same type.

Reply:
Yes we can stick with int too.